### PR TITLE
[orchagent]: Remove global lock caused by notifications running in another thread

### PR DESCRIPTION
**Pull request by pavel-shirshov**
**What I did**
I removed the notification handlers which were controlled by libsairedis and added direct support of notifications from ASIC_DB. 

**Why I did it**
To remove global lock from orchagent.

**How I verified it**
Install into the switch and run it.
Also /var/log/syslog shows notifications from port status change.

### CODE REVIEW

File: orchagent/orch.h [hunk 1]
```diff
@@ -152,7 +152,7 @@ class Orch
 
     /* Run doTask against a specific executor */
     virtual void doTask(Consumer &consumer) = 0;
-    virtual void doTask(NotificationConsumer &consumer) { }
+    virtual void doTask(NotificationConsumer &consumer, const std::string &name) { }
```
Comment (lguohan) [inline orchagent/orch.h]: >const std::string &name [](start = 56, length = 23)

can you add a comment on the name, what does this name use for?

File: orchagent/fdborch.cpp [hunk 2]
```diff
@@ -25,8 +26,14 @@ FdbOrch::FdbOrch(DBConnector *db, string tableName, PortsOrch *port) :
 {
     m_portsOrch->attach(this);
     auto consumer = new NotificationConsumer(db, "FLUSHFDBREQUEST");
-    auto fdbNotification = new Notifier(consumer, this);
+    auto fdbNotification = new Notifier(consumer, this, "FLUSHFDBREQUEST");
```
Comment (qiluo-msft) [inline orchagent/fdborch.cpp]: >fdbNotification [](start = 9, length = 15)

Confused about fdbNotification vs fdb_notification. What is the diff?

File: orchagent/notifier.h [hunk 3]
```diff
@@ -2,8 +2,8 @@
 
 class Notifier : public Executor {
 public:
-    Notifier(NotificationConsumer *select, Orch *orch)
-        : Executor(select, orch)
+    Notifier(NotificationConsumer *select, Orch *orch, const std::string& notifier_name)
```
Comment (qiluo-msft) [inline orchagent/notifier.h]: >notifier_name [](start = 74, length = 13)

I notice there is a similar name at NotificationConsumer.m_channel. I prefer using that one instread of a new parameter.

File: orchagent/fdborch.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (pavel-shirshov) [reply orchagent/fdborch.cpp]: With enumerations we will have hardcoded enumerations.
I think in this case it's better to develop better abstraction than NotificationConsumer(). Currently it's connected too deep into Orch class (doTask methods for NotificationConsumer). We should decouple them from each other and implement doTask methods as a base class for Orch classes.

File: orchagent/orch.h (hunk 1 above)
Comment (pavel-shirshov) [reply orchagent/orch.h]: Comments added

Comment (pavel-shirshov) [reply orchagent/orch.h]: I've also renamed this parameter to more clear name

File: orchagent/fdborch.cpp (hunk 2 above)
Comment (pavel-shirshov) [reply orchagent/fdborch.cpp]: Renamed all the variable names for notification related stuff

File: orchagent/notifier.h (hunk 3 above)
Comment (pavel-shirshov) [reply orchagent/notifier.h]: Renamed to m_channel

File: orchagent/fdborch.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (zhenggen-xu) [reply orchagent/fdborch.cpp]: I agree more changes is needed for a better abstraction than what we have today. but I am not sure if you are going to do it in this PR.
My point was, at least we should put the message types in a common place and reuse then on different places so you don't end up with mistype something and get run-time problem.  enumerations will detect the errors during the build time not run time if you mistyped, also by putting things in the same place improves manageability and extendability.

**Pull request merged**